src/main/java/ca/corefacility/bioinformatics/irida/repositories/filesystem/IridaFileStorageLocalUtilityImpl.java

```diff
@@ -220,4 +220,48 @@ public String getFileExtension(List<? extends SequencingObject> sequencingObject
 		}
 		return bytes;
 	}
+
+	@Override
+	public boolean checkConnectivity() {
+		// Since it's the local filesystem we just return true
+		return true;
+	}
+
+	@Override
+	public boolean checkWriteAccess(Path baseDirectory) {
+		if (!Files.exists(baseDirectory)) {
+			logger.error("Cannot continue startup; base directory " + baseDirectory + " does not exist!");
+			System.exit(1);
+		} else {
+			try {
+				// Check if basedirectory path is writeable by creating a temp file and then removing it
+				Path tempFile = Files.createTempFile(baseDirectory, "", "");
+				// Check if directory is writeable
+				boolean directoryWriteable = Files.isWritable(tempFile);
+
+				try {
+					// Cleanup the temp file created in the directory
+					Files.delete(tempFile);
+				} catch (IOException e) {
+					logger.error("An I/O error occurred while attempting to remove temp file ", e);
+				}
+
+				if (!directoryWriteable) {
+					// Log the error and exit so startup does not continue
+					logger.error("Cannot continue startup; base directory " + baseDirectory
+							+ " does not have write access! Please check directory permissions.");
+					System.exit(1);
```

Comment:
I don't think we want to directly call system.exit in any of the cases in here.  It should more likely throw an exception that doesn't get caught.  That way the servlet container (tomcat) decides how to handle things.  Can the `System.exit` lines get refactored out in all the cases and just replaced with a throw (maybe `StorageException`).

Reply:
Refactored to use StorageException in b47ef804ace2d29d02bac15593fc7f7e5129e205